tests/all/gc.rs

```diff
@@ -215,373 +202,226 @@ fn drop_externref_via_table_set() -> anyhow::Result<()> {
         "#,
     )?;
 
-    let instance = Instance::new(&store, &module, &[])?;
-    let table_set = instance.get_func("table-set").unwrap();
+    let instance = Instance::new(&mut store, &module, &[])?;
+    let table_set = instance.get_func(&mut store, "table-set").unwrap();
 
-    let foo_is_dropped = Rc::new(Cell::new(false));
-    let bar_is_dropped = Rc::new(Cell::new(false));
+    let foo_is_dropped = Arc::new(AtomicBool::new(false));
+    let bar_is_dropped = Arc::new(AtomicBool::new(false));
 
     let foo = ExternRef::new(SetFlagOnDrop(foo_is_dropped.clone()));
     let bar = ExternRef::new(SetFlagOnDrop(bar_is_dropped.clone()));
 
     {
         let args = vec![Val::ExternRef(Some(foo))];
-        table_set.call(&args)?;
+        table_set.call(&mut store, &args)?;
     }
     store.gc();
-    assert!(!foo_is_dropped.get());
-    assert!(!bar_is_dropped.get());
+    assert!(!foo_is_dropped.load(SeqCst));
+    assert!(!bar_is_dropped.load(SeqCst));
 
     {
         let args = vec![Val::ExternRef(Some(bar))];
-        table_set.call(&args)?;
+        table_set.call(&mut store, &args)?;
     }
     store.gc();
-    assert!(foo_is_dropped.get());
-    assert!(!bar_is_dropped.get());
+    assert!(foo_is_dropped.load(SeqCst));
+    assert!(!bar_is_dropped.load(SeqCst));
 
-    table_set.call(&[Val::ExternRef(None)])?;
-    assert!(foo_is_dropped.get());
-    assert!(bar_is_dropped.get());
+    table_set.call(&mut store, &[Val::ExternRef(None)])?;
+    assert!(foo_is_dropped.load(SeqCst));
+    assert!(bar_is_dropped.load(SeqCst));
 
     Ok(())
 }
 
 #[test]
-fn gc_in_externref_dtor() -> anyhow::Result<()> {
-    let (store, module) = ref_types_module(
-        r#"
-            (module
-                (table $t 1 externref)
+fn global_drops_externref() -> anyhow::Result<()> {
+    test_engine(&Engine::default())?;
+    test_engine(&Engine::new(Config::new().allocation_strategy(
+        InstanceAllocationStrategy::Pooling {
```

Comment:
Nit: we can replace this (and below) with `InstanceAllocationStrategy::pooling()` for brevity.